domain/bookings.py

```diff
@@ -25,3 +48,56 @@ def filter_bookings_to_compute_remaining_stock(stock: Stock) -> Iterator:
                                 and b.dateUsed
                                 and b.dateUsed >= stock.dateModified),
                   stock.bookings)
+
+
+def generate_qr_code(booking: Booking) -> str:
+    qr = qrcode.QRCode(
+        version=QR_CODE_VERSION,
+        error_correction=qrcode.constants.ERROR_CORRECT_Q,
+        box_size=QR_CODE_BOX_SIZE,
+        border=QR_CODE_BOX_BORDER,
+    )
+
+    offer_formula = ''
+    if booking.stock.offer.type == str(EventType.CINEMA):
+        offer_formula = 'PLACE'
+    if booking.stock.offer.type == str(ThingType.CINEMA_ABO):
+        offer_formula = 'ABO'
+
+    offer_type = 'EVT' if booking.stock.offer.isEvent else 'BIEN'
+    offer_date_time = booking.stock.beginningDatetime if booking.stock.beginningDatetime else ''
+    product_extra_data = booking.stock.offer.product.extraData
```

Comment:
Vu que l'on peut modifier l'ISBN sur une offre, il faudrait plutôt prendre l'extradata de offer.
```suggestion
    offer_extra_data = booking.stock.offer.extraData
```

Reply:
👍